eng/liveILLink.targets

```diff
@@ -36,8 +36,8 @@
                       Private="false"
                       SetConfiguration="Configuration=$(ToolsConfiguration)">
         <!-- Keep TFMs in sync with ILLink.Tasks.csproj -->
-        <SetTargetFramework Condition="'$(MSBuildRuntimeType)' == 'Core'">TargetFramework=$(NetCoreAppToolCurrent)</SetTargetFramework>
-        <SetTargetFramework Condition="'$(MSBuildRuntimeType)' != 'Core'">TargetFramework=$(NetFrameworkToolCurrent)</SetTargetFramework>
+        <SetTargetFramework Condition="'$(MSBuildRuntimeType)' == 'Core'">TargetFramework=$(TargetFrameworkForNETCoreTasks)</SetTargetFramework>
+        <SetTargetFramework Condition="'$(MSBuildRuntimeType)' != 'Core'">TargetFramework=$(TargetFrameworkForNETFrameworkTasks)</SetTargetFramework>
```

Comment:
I don't think these changes are an improvement. We prefer the `NetCoreAppToolCurrent` and `NetFrameworkToolCurrent` properties over `TargetFrameworkForNETCoreTasks` and `TargetFrameworkForNETFrameworkTasks`. I don't understand why the latter exist at all.

My plan is to eventually replace our custom properties in runtime with the standardized ones from Arcade: https://github.com/dotnet/arcade/blob/2d7ad177f759e8a0128cc53fa90ffc54aec1822c/src/Microsoft.DotNet.Arcade.Sdk/tools/TargetFrameworkDefaults.props#L12

Reply:
hmm I thought there was some case where NetCoreAppToolCurrent and TargetFrameworkForNETCoreTasks didn't align but I can't remember anymore.

Reply:
I'd prefer to leave this as is for now so we get consistency across the msbuild tasks and then clean it up later.